e2e/playwright/special-permissions-actions-available/src/tests/special-permissions-actions.test.ts

```diff
@@ -0,0 +1,190 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as testData from '@alfresco/playwright-shared';
+import { viewerTests } from './viewer';
+import {
+  ApiClientFactory,
+  FavoritesPageApi,
+  FileActionsApi,
+  TEST_FILES,
+  NodesApi,
+  SitesApi,
+  test,
+  SharedLinksApi,
+  SearchPageApi
+} from '@alfresco/playwright-shared';
+import { Site } from '@alfresco/js-api';
+
+test.describe('Special permissions : ', () => {
+  const apiClientFactory = new ApiClientFactory();
+  const random = testData.random;
+  let docLibId: string;
+
+  const sitePrivate = `site-private-${random}`;
+
+  const userManager = `manager-${random}`;
+  const userConsumer = `consumer-${random}`;
+  const userCollaborator = `collaborator-${random}`;
+  const userDemoted = `demoted-${random}`;
+
+  let fileDocxFavId: string;
+  let fileFavId: string;
+  let fileDocxSharedId: string;
+  let fileDocxSharedFavId: string;
+  let fileSharedId: string;
+  let fileSharedFavId: string;
+  let fileLockedId: string;
+  let fileFavLockedId: string;
+  let fileSharedLockedId: string;
+  let fileSharedFavLockedId: string;
+  let fileGranularPermissionId: string;
+  let fileLockedByUserId: string;
+  let folderFavId: string;
+  let folderFav2Id: string;
+
+  let managerNodeActions: NodesApi;
+  let demotedUserActions: NodesApi;
```

Comment:
Do we really need to have separate api object for each user required in the tests? Maybe we can have just one object per api type and just authenticate with different user when required? @azakrzewski-hy what do you think?

Reply:
I guess it will complicate the setup process as we need to authenticate 1-2 times for each case.